discord/template.py

```diff
@@ -150,18 +153,18 @@ def _store(self, data: TemplatePayload):
             source_serialised = data['serialized_source_guild']
             source_serialised['id'] = id
             state = _PartialTemplateState(state=self._state)
-            guild = Guild(data=source_serialised, state=state)  # type: ignore
+            guild = Guild(data=source_serialised, state=state)  # type: ignore - Guild expects a ConnectionState, we're passing a _PartialTemplateState
 
-        self.source_guild = guild
-        self.is_dirty = data.get('is_dirty', None)
+        self.source_guild: Guild = guild # type: ignore - id is a required field on the discord API docs
```

Comment:
I'm not sure I follow why an assignment needs to be `# type: ignore`'d.

Reply:
`self._state._get_guild(id)` returns `Unknown | None`. Instead of a `# type: ignore`, `and id` from `if guild is None and id` could also be removed because `guild` is set as a `Guild` instance in the if block.